### ISSUE TITLE
Broadwell - Problem adding packages.synocommunity.com repository to rs3617xs+

### ISSUE THREAD

**Issue by mukicz** [SynoCommunity/spksrc#2509]
Hello,
  it seems there is a problem adding packages.synocommunity.com repository to rs3617xs+ - it reports "Invalid location".  The same procedure on RS2211 works well.

Here are the http request and reply sniffed from the network ... (packages.synocommunity.com was resolved to 104.31.94.179)

```
GET /?package_update_channel=beta&unique=synology_broadwell_rs3617xs%2B&build=8451&language=csy&major=6&arch=broadwell&minor=0&timezone=Amsterdam HTTP/1.1
Host: packages.synocommunity.com
User-Agent: synology_broadwell_rs3617xs+ DSM6.0-8451 Update 2 (package)
Accept: */*

```

```
HTTP/1.1 422 UNPROCESSABLE ENTITY
Date: Wed, 02 Nov 2016 11:47:48 GMT
Content-Type: text/html
Transfer-Encoding: chunked
Connection: keep-alive
Set-Cookie: __cfduid=d1523eb3770083bbc6dad219f752ed3ab1478087268; expires=Thu, 02-Nov-17 11:47:48 GMT; path=/; domain=.synocommunity.com; HttpOnly
Server: cloudflare-nginx
CF-RAY: 2fb75a9440222348-FRA

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>422 Unprocessable Entity</title>
<h1>Unprocessable Entity</h1>
<p>The request was well-formed but was unable to be followed due to semantic errors.</p>

```
(and one more POST request using the same parameters)

**Comment by piwi82** [comment]
The new architecture `broadwell` is not yet recognized par the repository ([not listed in the Wiki](https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model)).

**Comment by KaraokeStu** [comment]
Yup, and there's currently no source from synology for this architecture either, so without that, we can't compile the right kernel

**Comment by Dr-Bean** [comment]
Broadwell is x64.

**Comment by piwi82** [comment]
The `broadwell`  architecture should be added [here](https://github.com/SynoCommunity/spksrc/blob/master/mk/spksrc.common.mk#L41).

**Comment by Matioupi** [comment]
Hello, 
same issue here; i've not been able to add the synocommunity package source on a RS3617xs+ running DSM 6.0.2 8451 update 4
I've tried either packages.synocommunity.com and www.synocommunity.com with http or https.
In each case, I got a "emplacement incorrect" error message (french language)
On the same LAN and DSM version, a DS1812+ got no issue.
DNS seems to be working properly on the RS3617xs+ but there is still something strange 👍 when loggued in SSH, I can ping a url when root (sudo -i) but not from admin account.
Could it be that name resolution from admin user is restricted ?

**Comment by Matioupi** [comment]
broadwell architecture source code is now available from Synology Open Source :

https://sourceforge.net/projects/dsgpl/files/Synology%20NAS%20GPL%20Source/

the toolkit is still missing...

**Comment by Dr-Bean** [comment]
I've added the Broadwell arch to the repository. Note that we haven't uploaded any packages yet, so at this point you can only add the repository to your NAS. The only packages you'll see are packages set to `noarch`.

Packages will be uploaded once the DSM6 toolchains are added, and the reason that hasn't happened yet is related to #2216, and the missing kernel files.

**Comment by aamri111** [comment]
Any luck here on supporting "broadwell" architecture? I upgraded to a DS3617xs :( 
@piwi82

**Comment by piwi82** [comment]
I don't maintain spksrc. You should ask @SynoCommunity/developers if someone has time to add this new architecture.

**Comment by aamri111** [comment]
Thanks

**Comment by KaraokeStu** [comment]
As stated, there are currently no packages available for this yet, other than ones set to **noarch** once we have the new toolchains (which have been promised from Synology for early March 2017) we should be able to start making traction on this

**Comment by ymartin59** [comment]
@KaraokeStu @Dr-Bean Tool chains for broadwell platform has been published for both DSM 6.0.2 and DSM 6.1: https://sourceforge.net/projects/dsgpl/files/

**Comment by ymartin59** [comment]
@KaraokeStu @Dr-Bean I have submitted PR #2727 with DSM 6.1 toolchains, including broadwell platform. What should be done now to get this arch supported in package repository ?

**Comment by gierran** [comment]
DS3617xs here - also very hopeful to get the missing packages added for broadwell. :)

**Comment by spacekiek** [comment]
Hi 

Any update for the broadwell platform?
Got a brand new DS3617xs, but I can only see the noarch of course ...

**Comment by PCDesmond** [comment]
Hey guys,

Also waiting for updates on the broadwell packages, any updates?

**Comment by rnsc** [comment]
ditto, if I can be of any help with the devices I have, let me know too
not sure how I can help move things forward though.

**Comment by gierran** [comment]
+1

**Comment by m4tt075** [comment]
@Dr-Bean Any free time on your horizon where you will be able to continue the DSM6/6.1 transition you have started?

**Comment by koljaschluetter** [comment]
Push.

**Comment by spacekiek** [comment]
Hi ... can we be of some assistance here?
Really would like to install your packages on broadwell Arc

**Comment by MtechD** [comment]
Now I finally understand why my DS3617xs isn't showing all packages  ! 

Can we expect an update soon for Broadwell  ?

**Comment by spacekiek** [comment]
Hello

Don't mean to be disrespectful, but will Broadwell ever be supported ? 
I'm checking every day whether the packages will show up on my DS3617xs :-)

BR

**Comment by cytec** [comment]
see: https://github.com/SynoCommunity/spksrc/issues/2509#issuecomment-267825167 or https://github.com/SynoCommunity/spksrc/issues/2216 broadwell will be supported when DSM 6 Support is done

**Comment by PCDesmond** [comment]
Hasn't DSM 6 been supported for a while now though?  I can download and run SynoCommunity packages to my DS1815+ running DSM 6.1 already.  Or is the issue specific to Broadwell?

**Comment by ymartin59** [comment]
Packages available in repository have been built for DSM 4.x or 5.x and some of them (happily) still work well on DSM 6.x. As I cannot produce DSM 5.2 package for Broadwell, this architecture has to wait for full DSM 6.x support in repository (spkrepo)

**Comment by ymartin59** [comment]
I may try to produce two or three packages for Broadwell and publish them here for a manual installation. So choose your package set you would request well...

**Comment by aamri111** [comment]
@ymartin59 Thaaaaaaaanks please CP, sonarr, NZBget

**Comment by rnsc** [comment]
Well Debian chroot for Broadwell would be awesome.
And again, I have hardware I'm willing to test things on if needed.

**Comment by ymartin59** [comment]
CouchPotato, Sonarr and NZBget (#2880) face same issue with service account creation/management: #2216
Debian chroot is even more complex... DSM docker support (limited to x86_64) is an alternative for most these services....

**Comment by ymartin59** [comment]
Request for testing: I have deployed ffmpeg, chromaprint and mediainfo for Broadwell. May you confirm these packages are visible (and maybe usage...) ?

**Comment by PCDesmond** [comment]
As of now, none of them show up in Package Center on my DS3617xs.

**Comment by spacekiek** [comment]
Me neither, also a DS3617xs

**Comment by spacekiek** [comment]
Hello,

Are there any advances made for bringing more packages to Broadwell Arc ?

**Comment by gddabe** [comment]
https://drive.google.com/open?id=0B76jv3fDYwcvN2xMVXVzZzN3V2M
compiled one for DS3617xs, tested with positive result, manually installed to yours, cheers :)

**Comment by spacekiek** [comment]
How do you manually compile for DS3617xs ? 

I'd like to add several packages ....

**Comment by roysbike** [comment]
I compile ffmpeg for broadwell DSM 6.1 

[ffmpeg_broadwell-6.1_3.3.3-7.spk.zip](https://github.com/SynoCommunity/spksrc/files/1624956/ffmpeg_broadwell-6.1_3.3.3-7.spk.zip)

**Comment by Safihre** [comment]
Test packages for DSM6.1 and new architectures are available for testing here #3138.
We hope to soon publish these on the official repo, which now should have support for `broadwell`.
